main/src/ca/uwaterloo/flix/util/Validation.scala

```diff
@@ -157,6 +185,9 @@ object Validation {
     for (x <- xs) {
       f(x) match {
         case Success(v) => successValues += v
+        case SuccessWithFailures(v, e) =>
+          successValues += v
+          failureStream += e
```

Comment:
We need a bit more tracking here. If we only have `SuccessWithFailures` then the whole traversal should be `SuccessWithFailures` rather than `Failure`

Reply:
Ah yes!